tests/test_streaming.py

```diff
@@ -85,6 +85,60 @@ def test_new_defaults_warning(local_remote_dir: Tuple[str, str], caplog: Callabl
     assert 'Because `num_canonical_nodes` was not specified,' in caplog.text
 
 
+@pytest.mark.parametrize('batch_size', [4, 7])
```

Comment:
There are many combinations (2 * 2 * 2 * 4), which increases the unit test time; do we need all of them? Can we reduce the number of combinations?

Reply:
I think it is ok to keep the combinations, they're all relevant. One improvement here may be to deduplicate the dataset creation.

Reply:
it's a very short test, because there's no dataloader its very quick. i'd like to keep.

but yeah there's future work to have fixtures for dataset creation since a lot of these tests use the same datasets.